autowarefoundation/autoware.universe
# fix(system_monitor): extend command line to display

## Description

- The process_monitor provides information about processes that are utilizing high CPU and memory resources.
However, the displayed COMMAND field only shows the executable name such as `ros2` or `python3`, and identifying the process solely based on the provided COMMAND name is not easy.
![image](https://github.com/autowarefoundation/autoware.universe/assets/57388357/8bc1c3ce-3642-4fcf-bb67-0c7bcabf6779)

- With this change, it's possible to provide a more detailed COMMAND string.

## Related links

[TIER IV INTERNAL LINK TO JIRA](https://tier4.atlassian.net/browse/RT0-28722)

## Tests performed

1. Run Autoware.
2. Run rqt_runtime_monitor.
    ```console 
    ros2 run rqt_runtime_monitor rqt_runtime_monitor
    ```
3. Verify a more detailed COMMAND string is provided.
![image](https://github.com/autowarefoundation/autoware.universe/assets/57388357/8be999bd-2e6b-4bf3-b039-1fd8d022fd5d)

## Notes for reviewers

None

## Interface changes

None

## Effects on system behavior

None

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.


## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.


## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.


After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/
